enhancements/support-for-realtime-kernel.md

```diff
@@ -0,0 +1,293 @@
+
+Weyland-Yutani Corp. wants to deploy OCP at their site for processing satellite signals
+from potential alien spacecraft.  They want to avoid as much down time as possible when
+deploying their cluster with real-time kernel support.  They generate install manifests
+via the OpenShift intsaller and include a MachineConfig with `kernelType: realtime` for
+their RHCOS nodes.
+
+#### Story 2 - Fresh Install (Day 2)
+
+Cyberdyne Systems wants to deploy OCP at their radio tower to handle the processing
+of radio signals.  Processing the signals requires the determinism and guarantees
+that come with a real-time kernel.  They deploy OCP succesfully and then create a
+MachineConfig to select the real-time kernel on their RHCOS nodes.
+
+#### Story 3 - Upgrade Path
+
+Nakatomi Corp. has an older OCP cluster deployed using all RHCOS nodes.  They plan on
+introducing a workload that would benefit from the use of the real-time kernel.
+They upgrade their OCP cluster to the latest version with real-time kernel support
+and create a MachineConfig that selects the real-time kernel on their RHCOS nodes.
+
+#### Story 4 - Node Replacement
+
+Tyrell Corp. has an existing OCP cluster deployed with RHEL 7 worker nodes using
+the real-time kernel.  They want to upgrade to the latest version of OCP and switch
+their worker nodes to RHCOS nodes.  They remove the RHEL 7 worker nodes from the
+cluster, upgrade the cluster, add in RHCOS nodes, and create a MachineConfig to
+select the real-time kernel on their RHCOS nodes.
+
+#### Story 5 - Unsupported Version
+
+Corellian Engineering Corp. has an existing OCP cluster that does not support
+real-time kernels.  They learn of support for real-time kernels in the new version
+of OCP and try to create a MachineConfig that selects the real-time kernel.  The
+MachineConfig is ignored.
+
+### Implementation Details/Notes/Constraints
+
+This proposal only covers providing the real-time kernel to RHCOS nodes and
+selecting it via MachineConfig.  It **DOES NOT** cover any changes that may be
+required by the container runtime or other components of OCP.
+
+### Risks and Mitigations
+
+#### Performance Impact
+
+Selecting the real-time kernel for RHCOS nodes without properly configuring the
+underlying hardware **MAY** incur an undesirable performance impact.  It is
+recommended that only users who control their hardware (i.e. not cloud users)
+attempt to use the real-time kernel.  If users enable real-time kernel support
+without properly configured hardware, they can always remove the offending
+MachineConfig to return to the previous state.
+
+#### Lack of Full Support
+
+If there are additional changes required across the OCP product to properly support
+real-time kernel (i.e. container runtime, etc) which cannot be delivered at the
+same time, the real-time kernel packages can still be included as part of the
+`machine-os-content` image.  Additionally, exposing the tunable in the MachineConfig
```

Comment:
Mind clarifying this? The part about lacking full support pivots to removing the tunable.

Reply:
My logic was:

1) we prep RT kernel support in RHCOS + MCO
2) we discover there are other components that are not ready
3) mitigation:  keep the RT kernel in the `machine-os-content` and backout the MCO change, effectively hiding the knob from users

Does that help?